### PR TITLE
clip at half resolution option

### PR DESCRIPTION
**Pull request by whyboris**
🎉 Video can now be extracted at half resolution 🎉 

This should be very welcome for when extracting screenshots at high resolutions 🤤

### CODE REVIEW

**Comment by whyboris** [comment]
I tested at various resolutions -- but didn't try many file types -- I'm afraid _something_ might sometimes fail 🤷‍♂ 

@cal2195 if you have, off the top of your head, any concerns, please comment sometime 🙇

File: main-extract.ts [hunk 1]
```diff
@@ -276,7 +277,8 @@ const generatePreviewClip = (
   screenshotHeight: number,
   saveLocation: string,
   clipSnippets: number,
-  snippetLength: number
+  snippetLength: number,
+  clipHalfRez: boolean
```
Comment (cal2195) [inline main-extract.ts]: fo shiz

File: main-extract.ts [hunk 2]
```diff
@@ -300,7 +302,8 @@ const generatePreviewClip = (
       concat += '[' + (current - 1) + ':V]' + '[' + (current - 1) + ':a]';
       current++;
     }
-    concat += 'concat=n=' + (totalCount - 1) + ':v=1:a=1[v][a];[v]scale=-2:' + screenshotHeight + '[v2]';
+    concat += 'concat=n=' + (totalCount - 1) + ':v=1:a=1[v][a];[v]scale=-2:' +
+              (clipHalfRez ? screenshotHeight / 2 : screenshotHeight) + '[v2]';
```
Comment (cal2195) [inline main-extract.ts]: I assume ffmpeg will handle it, but have to tried halving an _odd_ number?

**Comment by cal2195** [comment]
One thought - on a large library, people might have _small_ resolutions and _large_ resolutions. In this case, halving the small rezzzzzes would be annoying, but not halving the large ones would be bad too - maybe having a `MAX_HEIGHT` would be better instead?

File: main-extract.ts (hunk 2 above)
Comment (whyboris) [reply main-extract.ts]: Currently I've hardcoded these heights:
```
'144', '216', '288', '360', '432'
```
No other ones are allowed. Halving any of them seems to work fine when I did spot-testing in the app 😅

**Comment by whyboris** [comment]
I'm unsure I can picture what you're describing exactly. Please let me know if the suggestion below misses the point.

I guess there's no difficulty in having a separate selector for the _clipResolution_ rather than just a dumb _boolean_ toggle for _halfRez_ VS the default full resolution. 

So I'm thinking rather than `MAX_HEIGHT` we'll just have the `ScreenshotSettings` have `clipHeight` in addition to `height` 🤷‍♂

**Comment by whyboris** [comment]
Improved functionality with #265 -- can manually select clip height 👍

**Pull request merged**